src/PersistentFilters.ts

```diff
@@ -160,68 +130,146 @@ export class PersistentFilters {
         this.mSessions.sort();
         this.updateSessions();
     }
-    public async removeSession(criteria: string) {
+
+    /*********************************************************************************************************************************************/
+    /* Get/read functions, for returning the values stored in the persistent arrays
+    /*********************************************************************************************************************************************/
+
+    /**
+     * Returns the current contents of the persistent search history array
+     *
+     * @returns {string[]} persistent array of search history items
+     */
+    public getSearchHistory() {
+        return this.mSearchHistory;
+    }
+
+    /**
+     * Returns the current contents of the persistent sessions array
+     *
+     * @returns {string[]} persistent array of sessions and their settings
+     */
+    public getSessions() {
+        return this.mSessions;
+    }
+
+    /**
+     * Returns the current contents of the persistent recently-opened file history array
+     *
+     * @returns {string[]} persistent array of recently-opened files
```

Comment:
We should not duplicate the type information in JSDoc.
```suggestion
     * @returns persistent array of recently-opened files
```